docs/changes/modeling/12411.api.rst

```diff
@@ -0,0 +1,2 @@
+Deprecated the ``AliasDict`` class, removed the ``_parameter_with_unit``,
+and removed the ``_parameter_without_unit`` method from ``modeling.utils``.
```

Comment:
Should the removal of private functions be announced in the change log?

Reply:
Removed mentions of `_parameter_with_unit` and `_parameter_without_unit` from changelog entry in e9c0acb.